processor/groupbytraceprocessor/factory.go

```diff
@@ -0,0 +1,107 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package groupbytraceprocessor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configerror"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer"
+)
+
+const (
+	// typeStr is the value of "type" for this processor in the configuration.
+	typeStr configmodels.Type = "groupbytrace"
+)
+
+var (
+	defaultWaitDuration   = time.Second
+	defaultNumTraces      = 1_000_000
+	defaultDiscardOrphans = false
+	defaultStoreInDisk    = false
+
+	errDiskStorageNotSupported    = fmt.Errorf("option 'disk storage' not supported in this release")
+	errDiscardOrphansNotSupported = fmt.Errorf("option 'discard orphans' not supported in this release")
+)
+
+// Factory is the factory for the processor.
+type Factory struct {
+}
+
+var _ component.ProcessorFactory = (*Factory)(nil)
+
+// Type gets the type of the config created by this factory.
+func (f *Factory) Type() configmodels.Type {
+	return typeStr
+}
+
+// CreateDefaultConfig creates the default configuration for the processor.
+func (f *Factory) CreateDefaultConfig() configmodels.Processor {
+	return &Config{
+		ProcessorSettings: configmodels.ProcessorSettings{
+			TypeVal: typeStr,
+			NameVal: string(typeStr),
+		},
+		NumTraces:    defaultNumTraces,
+		WaitDuration: defaultWaitDuration,
+
+		// not supported for now
+		DiscardOrphans: defaultDiscardOrphans,
```

Comment:
I decided to keep these two out for this PR, as it was already big enough for a first proposal. Once this PR gets merged, I'll work on implementing these two features (discard orphans and store in disk).

Reply:
I suggest to remove these options and add them in appropriate PRs with corresponding functionality.

Reply:
I'll indeed remove them before this gets ready to be merged. Those options are here to show the direction this process is intended to go and to foster discussion :-)